homeassistant/components/honeywell/sensor.py

```diff
@@ -0,0 +1,109 @@
+"""Support for Honeywell (US) Total Connect Comfort sensors."""
+from __future__ import annotations
+
+from homeassistant.components.honeywell.const import (
+    DOMAIN,
+    SENSOR_LOCATION_INDOOR,
+    SENSOR_LOCATION_OUTDOOR,
+)
+from homeassistant.components.sensor import STATE_CLASS_MEASUREMENT, SensorEntity
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.device_registry as dr
+
+
+async def async_setup_entry(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Honeywell thermostat."""
+    data = hass.data[DOMAIN][config.entry_id]
+    sensors = []
+
+    if data.device.current_temperature is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_INDOOR, DEVICE_CLASS_TEMPERATURE)
+        )
+    if data.device.current_humidity is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_INDOOR, DEVICE_CLASS_HUMIDITY)
+        )
+    if data.device.outdoor_temperature is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_OUTDOOR, DEVICE_CLASS_TEMPERATURE)
+        )
+    if data.device.outdoor_humidity is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_OUTDOOR, DEVICE_CLASS_HUMIDITY)
+        )
+
+    async_add_entities(sensors)
+
+
+class HoneywellUSSensor(SensorEntity):
+    """Representation of a Honeywell US Sensor."""
+
+    def __init__(self, data, location, sensor_type):
+        """Initialize the sensor."""
+        self._data = data
+        self._attr_unique_id = (
+            f"{dr.format_mac(data.device.mac_address)}_{location}_{sensor_type}"
+        )
+        self._attr_name = f"{data.device.name} {location} {sensor_type}"
+        self._location = location
+        self._type = sensor_type
+
+    @property
+    def device(self):
+        """Shortcut to access the device."""
+        return self._data.device
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return self._type
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._type == DEVICE_CLASS_TEMPERATURE:
+            return self.temperature
+        if self._type == DEVICE_CLASS_HUMIDITY:
+            return self.humidity
+
+    @property
+    def state_class(self):
+        """Return the state class of the sensor."""
+        return STATE_CLASS_MEASUREMENT
```

Comment:
This can be set as a class attribute:

<https://developers.home-assistant.io/docs/core/entity/#entity-class-or-instance-attributes>